### ISSUE TITLE
Cannot install no matter the node version

### ISSUE THREAD

**Issue by sagecorbett** [cristovao-trevisan/pitchfinder#4]
I've tried switching Node to version 9 and still cannot seem to install. Any thoughts on what I could try? I've gone through many threads with a similar error regarding node-gyp but still no solution. 
```npm install node-pitchfinder
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated har-validator@5.1.5: this library is no longer supported

> node-pitchfinder@2.0.0 install /Users/sagecorbett/Desktop/Everything/REPOS/Music/node_modules/node-pitchfinder
> node-gyp rebuild

  CXX(target) Release/obj.target/addon/src/detectors/addons.o
  CXX(target) Release/obj.target/addon/src/detectors/yin/addon/yin.o
../src/detectors/yin/addon/yin.cc:21:3: warning: ignoring return value of function declared with 'warn_unused_result' attribute [-Wunused-result]
  exports->Set(
  ^~~~~~~~~~~~
../src/detectors/yin/addon/yin.cc:72:3: warning: ignoring return value of function declared with 'warn_unused_result' attribute [-Wunused-result]
  result->Set(context, Nan::New("pitch").ToLocalChecked(), Nan::New(pitch));
  ^~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/detectors/yin/addon/yin.cc:73:3: warning: ignoring return value of function declared with 'warn_unused_result' attribute [-Wunused-result]
  result->Set(context, Nan::New("probability").ToLocalChecked(), Nan::New(obj->probability));
  ^~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
3 warnings generated.
  CXX(target) Release/obj.target/addon/src/detectors/macleod/addon/macleod.o
../src/detectors/macleod/addon/macleod.cc:25:3: warning: ignoring return value of function declared with 'warn_unused_result' attribute
      [-Wunused-result]
  exports->Set(context, Nan::New("MacLeod").ToLocalChecked(), tpl->GetFunction(context).ToLocalChecked());
  ^~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/detectors/macleod/addon/macleod.cc:81:3: warning: ignoring return value of function declared with 'warn_unused_result' attribute
      [-Wunused-result]
  result->Set(context, Nan::New("pitch").ToLocalChecked(), Nan::New(pitch));
  ^~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/detectors/macleod/addon/macleod.cc:82:3: warning: ignoring return value of function declared with 'warn_unused_result' attribute
      [-Wunused-result]
  result->Set(context, Nan::New("probability").ToLocalChecked(), Nan::New(obj->probability));
  ^~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
3 warnings generated.
  SOLINK_MODULE(target) Release/addon.node
ld: library not found for -latomic
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Release/addon.node] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/sagecorbett/.nvm/versions/node/v9.11.2/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at ChildProcess.emit (events.js:180:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:12)
gyp ERR! System Darwin 20.5.0
gyp ERR! command "/Users/sagecorbett/.nvm/versions/node/v9.11.2/bin/node" "/Users/sagecorbett/.nvm/versions/node/v9.11.2/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/sagecorbett/Desktop/Everything/REPOS/Music/node_modules/node-pitchfinder
gyp ERR! node -v v9.11.2
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok 
npm WARN music@1.0.0 No description

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! node-pitchfinder@2.0.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the node-pitchfinder@2.0.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/sagecorbett/.npm/_logs/2021-06-02T19_43_22_042Z-debug.log```

**Comment by cristovao-trevisan** [comment]
The problem is as said in your logs:
```
ld: library not found for -latomic
```

You need to install libatomic if not installed already.

Why is it needed?? I honestly don't know, probably some internal dependency of node-gyp

**Comment by cristovao-trevisan** [comment]
If you can, please provide information about you hardware and OS

**Comment by sagecorbett** [comment]
I am on a Mac OS running Big Sur (11.4). I also tried Mac OS Mojave. I am looking into what you suggested currently.

EDIT: No luck. Libatomic does not seem to be something a Mac OS can install and I cannot really find any information about it online.

**Comment by psherk-bw** [comment]
I'm getting the same issue as well, looks like `ld: library not found for -latomic` keeps popping up. I'm also on Mac OS running Big Sur

**Comment by cristovao-trevisan** [comment]
I don't have a Mac so I'm not able to reproduce this error, as it does install on Linux

**Comment by cristovao-trevisan** [comment]
For the time being I recommend you to use the pure JavaScript package ([npm install --save pitchfinder](https://github.com/peterkhayes/pitchfinder)).
It is slower but should run fine on decent hardware (should be true for a MacOS)

**Comment by cristovao-trevisan** [comment]
I also found some information online that recommends updating your XCode
If you can, also try to update the node version to the latest lts

**Comment by sagecorbett** [comment]
Yea no luck after updating everything. My Mac OS version is 11.4, I'm running Node 14, and XCode version 12.5. @psherk-bw were you able to find a solution? For now I am gonna check out the JS version. Thank you for your help @cristovao-trevisan

**Comment by chrisgloom** [comment]
same problem. Mac OS 10.14.6 
node -v
v14.17.6
npm -v
6.14.15
Will see if I can make any progress fixing myself.

**Comment by chrisgloom** [comment]
npm install completes when simply removing the library from bindings.gyp. Not sure to what extent that breaks functionality on an osx machine but i'd imagine it wouldn't if we don't have that library to begin with. I wonder if a solution there is to replace that library dependency with a script that checks if the current platform needs libatomic.

**Comment by cristovao-trevisan** [comment]
I just tested removing the libatomic dependency and it works fine on my linux, so I released a new version (2.1.0) that should work on MacOS (or any OS).

Please try and confirm it's working

**Comment by chrisgloom** [comment]
will check it out later today—thanks for looking into it!

**Comment by sagecorbett** [comment]
It's working for me! Thank you for the help!

**Comment by cristovao-trevisan** [comment]
Will be closing as fixed by #5
